src/rules/prefer-reactive-destructuring.ts

```diff
@@ -0,0 +1,64 @@
+import type { TSESTree } from "@typescript-eslint/types"
+import { createRule } from "../utils"
+
+export default createRule("prefer-reactive-destructuring", {
+  meta: {
+    docs: {
+      description: "Prefer destructuring from objects in reactive statements",
+      category: "Best Practices",
+      recommended: false,
+    },
+    hasSuggestions: true,
+    schema: [],
+    messages: {
+      useDestructuring: `Prefer destructuring in reactive statements`,
+      suggestDestructuring: `Use destructuring to get finer-grained redraws`,
+    },
+    type: "suggestion",
+  },
+  create(context) {
+    return {
+      // Finds: $: info = foo.info
+      // Suggests:  $: ({ info } = foo);
+      [`SvelteReactiveStatement > ExpressionStatement > AssignmentExpression[left.type="Identifier"][right.type="MemberExpression"]`](
```

Comment:
Isn't member access like the following checked by this rule? Doesn't it affect the number of redraws in the first place?

```svelte
$: infoBar = foo.info.bar
```

Reply:
Updated [the REPL](https://svelte.dev/repl/96759f4772314d0a840e11370ef76711) to show that `$: infoBar = foo.info.bar` style reactive statements will still re-render every time `foo` changes.

Reply:
Are you saying that changing to the following will reduce re-renders?

```js
$: ({ bar: infoBar } = foo.info)
```

If so, could you please add the following test case? I don't think the current selector checks that.

```js
$: infoBar = foo.info.bar
```

Reply:
I'm going to have it bail on multiply-nested rules and not make any suggestions, it's just too gross-looking. If the user wants to do the multiple-levels of destructuring more power to 'em.

Reply:
Thanks for adding the test cases.

> I don't think the current selector checks that.

I misread the selector 😓. It was already checked.